example/README.md

```diff
@@ -0,0 +1,52 @@
+# Running the example
+
+This [Docker Compose](.compose.yaml) file spins up a local environment for testing the agent. The environment starts two containers:
+
+- `redpanda-source`: simulates an IoT device that runs a single-node Redpanda instance and the agent to store and forward messages
+- `redpanda-destination`: simulates a central Redpanda cluster that aggregates messages from all of the IoT devices
+
+## Prerequisites
+
+1. The agent communicates with the source and destinations clusters over TLS enabled interfaces, so before starting the containers, run [generate-certs.sh](./generate-certs.sh) to create the necessary certificates. The resulting `./certs` directory is mounted on the containers
+2. Run the [build.sh](./build.sh) script to compile the agent for the Linux-based container
+
+## Testing the agent
+
+Open a terminal and produce some messages to the source's `telemetry` topic:
+
+```bash
+export REDPANDA_BROKERS=localhost:19092
+for i in {1..60}; do echo $(cat /dev/urandom | head -c10 | base64) | rpk topic produce telemetry; sleep 1; done
+```
+
+The agent will forward the messages to a topic with the same name on the destination container. Open a second terminal and consume the messages:
+
+```bash
+export REDPANDA_BROKERS=localhost:29092
```

Comment:
This is odd because the agent is configured to create the topics for you. I wonder if Redpanda's auto create topics configuration is set differently between the environments?

Reply:
Ahh I think I see the issue, I assumed the agent was running after following the prerequisite steps (`./generate-certs.sh` followed by `./build.sh`). I see now that the agent gets built but isn't started. I also see an `agent.yaml` file.

When trying to start with the existing agent file I get errors related to the default values. On Linux I also have an issue with file ownership, as I needed to run `sudo chown -R 101:101 certs` so that they would be able to be mounted within a docker container. But doing that makes these files inaccessible by the agent app. I'll instead modify folder permissions with chmod in order to get access and update the steps.

Reply:
Another issue I face is interrupting the agent. I initially didn't have Redpanda started when I first tested out the agent, so the agent started but couldn't connect to either the destination or source cluster. This caused a timeout for both that I wasn't able to break out of with CTRL-C.

More details on how to handle the interrupt signal are here: https://nathanleclaire.com/blog/2014/08/24/handling-ctrl-c-interrupt-signal-in-golang-programs/